OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyLogic.cs

```diff
@@ -26,7 +26,8 @@ public class LobbyLogic : ChromeLogic
 	{
 		static readonly Action DoNothing = () => { };
 
-		public MapPreview Map = MapCache.UnknownMap;
+		public MapPreview MapPreview = MapCache.UnknownMap;
+		public Map Map = null;
```

Comment:
Yes - you can more easily repro these crashes by simply returning from the worker threads without actually generating the previews.

Here's a few I can see offhand:
[LobbyMapPreviewLogic.cs#L28](https://github.com/pchote/OpenRA/blob/557fcbdedd012741498e10f910f3612d23f372fa/OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyMapPreviewLogic.cs#L28)
[LobbyMapPreviewLogic.cs#L59](https://github.com/pchote/OpenRA/blob/557fcbdedd012741498e10f910f3612d23f372fa/OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyMapPreviewLogic.cs#L59)
[LobbyLogic.cs#L193](https://github.com/pchote/OpenRA/blob/557fcbdedd012741498e10f910f3612d23f372fa/OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyLogic.cs#L193)
[LobbyLogic.cs#L320](https://github.com/pchote/OpenRA/blob/557fcbdedd012741498e10f910f3612d23f372fa/OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyLogic.cs#L320)